ci_watson/artifactory_helpers.py

```diff
@@ -124,6 +152,309 @@ def get_bigdata(*args):
     return dest
 
 
+def compare_outputs(outputs, raise_error=True, **kwargs):
+    """
+    Compare output with "truth" using appropriate
+    diff routine; namely,
+        ``fitsdiff`` for FITS file comparisons
+        ``unified_diff`` for ASCII products.
+
+    Only after all elements of `outputs` have been
+    processed will the method report any success or failure, with
+    failure of any one comparison *not* preventing the rest of the
+    comparisons to be performed.
+
+    Parameters
+    ----------
+    outputs : list of tuple or dicts
+        This list defines what outputs from running the test will be
+        compared.  Three distinct types of values as list elements
+        are supported::
+
+          - 2-tuple : (test output filename, truth filename)
+          - 3-tuple : (test output filename, truth filename, HDU names)
+          - dict : {'files':[], 'pars':()}
+
+        If filename contains extension such as '[hdrtab]' (no quotes),
+        it will be interpreted as specifying comparison of just that HDU.
+
+    raise_error : bool
+        Raise ``AssertionError`` if difference is found.
+
+    kwargs : keyword-value pairs
+        These user-specified inputs will use these values for the
+        parameters that control the operation of the diff
+        functions used in the comparison; namely, FITSDiff and HDUDiff.
+        This also includes the specification of the location of the
+        comparison files and where the results should be stored.
+        The currently supported attributes which can be overidden,
+        along with type of values accepted, includes::
+
+          - ignore_keywords : list
+          - ignore_hdus : list
+          - ignore_fields : list
+          - rtol : float
+          - atol : float
+          - results_root : string
+          - input_path : list
```

Comment:
Some explanation of these extra `kwargs` would be nice. For example, I cannot figure out what `results_root` is supposed to do without code diving.